### PR TITLE
Read user input properly in linux_install.sh for Gentoo

### PR DESCRIPTION
**Pull request by vomindoraan**
The script currently uses a very convoluted, hacky and error-prone way of reading user input (y/n) when installing on Gentoo, which involves changing TTY settings. This changes it to a POSIX-compliant `read` call.

It also replaces the non-POSIX `echo -n` with `printf`, as well as fixing some inconsistent indentation.

### CODE REVIEW

File: util/linux_install.sh [hunk 1]
```diff
@@ -75,13 +75,10 @@ elif grep ID /etc/os-release | grep -q 'arch\|manjaro'; then
 elif grep ID /etc/os-release | grep -q gentoo; then
 	echo GENTOO_WARNING | fmt
```
Comment (vomindoraan) [inline util/linux_install.sh]:
```suggestion
	echo $GENTOO_WARNING | fmt
```
Add missing `$`

File: util/linux_install.sh [hunk 2]
```diff
@@ -75,13 +75,10 @@ elif grep ID /etc/os-release | grep -q 'arch\|manjaro'; then
...
 	echo -n "Proceed (y/N)? "
```
Comment (vomindoraan) [inline util/linux_install.sh]:
```suggestion
	printf "Proceed (y/N)? "
```
Replace non-POSIX `echo -n` with `printf`

File: util/linux_install.sh [hunk 3]
```diff
@@ -68,20 +68,17 @@ elif grep ID /etc/os-release | grep -q 'arch\|manjaro'; then
...
-	answer=$( while ! head -c 1 | grep -i '[ny]' ;do true ;done )
-	stty $old_stty_cfg
-	if echo "$answer" | grep -iq "^y" ;then
+	echo "$GENTOO_WARNING" | fmt
+	printf "\nProceed (y/N)? "
+	read -r answer
+	if echo "$answer" | grep -iq "^y"; then
 		sudo touch /etc/portage/package.use/qmkfirmware
-		echo "sys-devel/gcc multilib" | sudo tee --append /etc/portage/package.use/qmkfirmware > /dev/null
+		echo "sys-devel/gcc multilib" >> /etc/portage/package.use/qmkfirmware
```
Comment (snortwolf) [inline util/linux_install.sh]: Creating the file as root and then echoing into it as a user isn't going to work.  Also, why is there even a prompt to continue for Gentoo when none of the other distros have this?

Comment (vomindoraan) [reply util/linux_install.sh]: Good catch, I omitted the `sudo`. I'll fix it right away.

Check out the [warning message](https://github.com/vomindoraan/qmk_firmware/blob/master/util/linux_install.sh#L5) at the top. The recently added Slackware distro also has a prompt for similar reasons.

Comment (vomindoraan) [reply util/linux_install.sh]:
```suggestion
		sudo echo "sys-devel/gcc multilib" >> /etc/portage/package.use/qmkfirmware
```

File: util/linux_install.sh [hunk 4]
```diff
@@ -78,7 +78,7 @@ elif grep ID /etc/os-release | grep -q gentoo; then
 	read -r answer
 	if echo "$answer" | grep -iq "^y"; then
 		sudo touch /etc/portage/package.use/qmkfirmware
-		echo "sys-devel/gcc multilib" >> /etc/portage/package.use/qmkfirmware
+		sudo echo "sys-devel/gcc multilib" >> /etc/portage/package.use/qmkfirmware
```
Comment (snortwolf) [inline util/linux_install.sh]: The shell redirection here is still invoked with user permissions, so it'll fail.

Comment (vomindoraan) [reply util/linux_install.sh]: Oh, so the pipe into `tee` is an idiom to get around that? TIL.

I'll revert my changes and add a comment to clarify that. Thanks again!

**Pull request merged**